src/components/sidenav/sidenav.js

```diff
@@ -231,6 +231,7 @@ function SidenavFocusDirective() {
  *
  * @param {expression=} md-is-open A model bound to whether the sidenav is opened.
  * @param {boolean=} md-disable-backdrop When present in the markup, the sidenav will not show a backdrop.
+ * @param {boolean=} md-disable-click-key-events When present in the markup, the click outside and 'ESC' button will not close the sidenav.
```

Comment:
Can we rename this to `md-disable-close-events`? So it would be possible to close via `toggle` or `open` functions, but not via events (backdrop click or escape key).

With a description like: "When present in the markup, clicking the backdrop or pressing the 'ESC' key will not close the sidenav."

Does this make sense? Do you have any ideas for a better API?

Reply:
Yes, it sounds good to me :).